### ISSUE TITLE
"Highlight" executable name is problematic for homebrew... please advise?

### ISSUE THREAD

**Issue by geoff-codes** [jgm/highlighting-kate#66]
Hi Prof —

I'm going to be submitting a [homebrew](https://github.com/Homebrew/homebrew) formula for highlighting-kate (specifically, the utility, since homebrew doesn't really manage Haskell libraries). However, there's a bit of an issue with the name of the utility. Because HFS+ is case-insensitive, "Highlight" conflicts with ["highlight"](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/highlight.rb), and I might also add that there's [npm's "highlight"](https://www.npmjs.com/package/highlight), GitHub's syntax highlighter, [highlights](https://github.com/atom/highlights), GNU source-highlight...

Presently, in the draft of the formula below, I'm renaming "Highlight" to "highlighting-kate". But I thought I'd ask if perhaps you'd prefer another name, or if this is something you would consider changing here "upstream"? Also, please let me know if you have any other thoughts or changes you'd like made in the formula!

```ruby
require "language/haskell"

class HighlightingKate < Formula
  include Language::Haskell::Cabal

  homepage "https://github.com/jgm/highlighting-kate"
  url "https://github.com/jgm/highlighting-kate/archive/0.5.14.tar.gz"
  sha256 "0c9ad39ee1b123b655434ff3a189c7aae634057711e120c08c01e4c310e959fd"

  head "https://github.com/jgm/highlighting-kate.git"

  depends_on "ghc" => :build
  depends_on "cabal-install" => :build
  depends_on "gmp"

  def install
    cabal_sandbox do
      cabal_install "--only-dependencies"
      system "make", "prep"
      cabal_install "--prefix=#{prefix}", "-fsplitBase", "-fexecutable"
    end
    cabal_clean_lib

    File.rename(bin/"Highlight", bin/"highlighting-kate")
  end

  test do
    system "highlighting-kate", "-s", "json", prefix/"INSTALL_RECEIPT.json"
  end
end
```

**Comment by jgm** [comment]
The new name seems fine.  Changing upstream probably makes
sense, too.

In the past homebrew always rejected things that could be
installed via cabal -- has that changed?

**Comment by geoff-codes** [comment]
> Changing upstream probably makes sense, too.

Great — see #68.


> In the past homebrew always rejected things that could be installed via cabal -- has that changed?

While I'm a little hesitant to give an unequivocal "yes" (I'm frequent homebrew contributor, but not a maintainer), homebrew's Haskell support has "grown up" quite a bit. Particularly, when cabal's sandboxing ability (where it often eliminates the need for the full "haskell-platform" install), programs written in Haskell (even those that are also Hackage packages) certainly **can** be [acceptable formulae](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Acceptable-Formulae.md#we-try-hard-to-avoid-dupes-in-homebrew). As cabal isn't really a full-fledged package manager (no real ability to uninstall or upgrade), homebrew and and cabal are often complementary when used in this fashion.

@mikemcquaid, anything to add? Maybe something on this for the docs?

**Comment by geoff-codes** [comment]
I'm going to go ahead and amend the above comment — it _is_ an unequivocal yes, if your [name is](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/pandoc.rb) [John](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/pandoc-citeproc.rb) [MacFarlane](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/texmath.rb). :wink:

**Comment by geoff-codes** [comment]
Closed by #68 and https://github.com/Homebrew/homebrew/pull/38963.
